Packages/com.unity.inputsystem/InputSystem/Controls/InputControlLayoutAttribute.cs

```diff
@@ -19,16 +19,16 @@ public class InputControlLayoutAttribute : Attribute
         /// state layout and you want the device layout to automatically take offsets from
         /// the fields annotated with <see cref="InputControlAttribute"/>.
         /// </remarks>
-        public Type stateType;
+        public Type stateType { get; set; }
 
-        public string stateFormat;
+        public string stateFormat { get; set; }
 
         ////TODO: rename this to just "usages"; "commonUsages" is such a weird name
-        public string[] commonUsages;
+        public string[] commonUsages { get; set; }
 
-        public string variants;
+        public string variants { get; set; }
 
-        public bool? updateBeforeRender;
+        internal bool? updateBeforeRender;
```

Comment:
This is a weird one - we have two public APIs for this, `updateBeforeRender`, and `beforeRender` - one is nullable and one is not. We should have only one. But I guess keeping `beforeRender` as I did also won't work, as it would throw if `updateBeforeRender` is null. What is this used for? Which ones do we need?

Reply:
Gah, took me a moment to decipher my mess there. Definitely missing a comment.

Looks like I added `beforeRender` because you can't initialize a nullable property through an attribute initializer. So yup, `internal` makes sense and I think it's on the right one.

The only thing I'm thinking is that it'd probably be better to swap the naming `beforeRender` <-> `updateBeforeRender` and let `beforeRender` be internal. And maybe actually call the latter one `updateBeforeRenderInternal` or something like that.